server/controllers/unfurl.js

```diff
@@ -418,31 +413,29 @@ const handleJoinedChannelEvent = async event => {
 
 const handleAppUninstalledEvent = async data => {
   // Do record clean up
-  try{
-  // get all users in this team
-  const users = await User.findAll({
-    where: { teamId: data.team_id }
-  });
-
-  await Promise.all(users.map(async user => {
-    // delete subscriptions for each user
-
-    // TODO : Should revoke access given to app by users in this workspace
-    // TODO : Should delete all DW active subscriptions by this user
-
-    // delete from DB
-    await Subscription.destroy({
-      where: { slackUserId: user.slackId }
-    });
-    // delete the user
-    await User.destroy({
-      where: { slackId: user.slackId }
+  try {
```

Comment:
I'm curious about this. I think it's great that we clean things up on our end. However, I'd like to make sure that if 1) we don't get this event or 2) something goes wrong and clean-up is partially complete, that there are no side-effects.

Specifically, I want to make sure that, if the app is reinstalled, it won't be in a corrupted state.

Reply:
We are performing 3 actions at that point. (edited)
1. delete all subscriptions for each user in the team
2. delete users  in the team
3. delete the team

If *1* fails or completes partially, When the app is re-installed the subscriptions not deleted will only be active and show up in slack if corresponding DW subscriptions are still active.

If *2* fails or completes partially,  When the app is re-installed the user records not deleted will be active if access has not been revoked from DW, if access had been revoked, account binding will be requested and user record will be updated

If *3* fails or completes partially,  When the app is re-installed the team record not deleted will get updated.